AK/Utf8View.cpp

```diff
@@ -224,14 +243,27 @@ u32 Utf8CodepointIterator::operator*() const
     size_t code_point_length_in_bytes = 0;
 
     bool first_byte_makes_sense = decode_first_byte(m_ptr[0], code_point_length_in_bytes, code_point_value_so_far);
-    if (!first_byte_makes_sense)
+
+    if (!first_byte_makes_sense) {
+        // The first byte of the codepoint doesn't make sense: output a replacement character
         dbgln("First byte doesn't make sense, bytes: {}", StringView { (const char*)m_ptr, m_length });
+        return 0xFFFD;
+    }
     VERIFY(first_byte_makes_sense);
```

Comment:
You do not need to `VERIFY` this because if it is false, the `if` above would have already returned.

Reply:
The VERIFY still makes sense : at this point of the code, we want to be sure that the first byte makes sense otherwise the code is invalid.

Rather than reaching an unrelated crash if somebody (somehow) changes the condition and we break an implicit VERIFY, I would rather make it explicit even if redundant and let the compiler optimize it away if needed !

Reply:
I see what you mean, but this argument could be made for every `if` statement that returns early then. But I guess that's a question of preference, so let's keep it that way.